src/js/proxified-containers.js

```diff
@@ -0,0 +1,148 @@
+// Below lets us print errors, huge thanks to Jonothan @ https://stackoverflow.com/questions/18391212/is-it-not-possible-to-stringify-an-error-using-json-stringify
+if (!("toJSON" in Error.prototype))
+  Object.defineProperty(Error.prototype, "toJSON", {
+    value: function() {
+      const alt = {};
+
+      Object.getOwnPropertyNames(this).forEach(function(key) {
+        alt[key] = this[key];
+      }, this);
+
+      return alt;
+    },
+    configurable: true,
+    writable: true
+  });
+
+
+// This object allows other scripts to access the list mapping containers to their proxies
+proxifiedContainers = {
+
+  // Slightly modified version of 'retrieve' which returns a direct proxy whenever an error is met.
+  retrieveFromBackground: function(cookieStoreId = null) {
+    return new Promise((resolve, reject) => {
+      proxifiedContainers.retrieve(cookieStoreId).then((success) => {
+        resolve(success.proxy);
+      }, function() {
+        resolve({
+          type: "direct"
```

Comment:
Another case where the constant could be used:
```suggestion
          type: Utils.DEFAULT_PROXY
```